shared/contracts/user.contract.ts

```diff
@@ -102,4 +103,13 @@ export const userContract = contract.router({
       400: contract.type<JSONAPIError>(),
     },
   },
+  resetPassword: {
+    method: 'POST',
+    path: '/users/me/password/reset',
```

Comment:
following the approach of moving the recover password flow to the auth controller given there is "no user" at that point, Ive decided to put this in within the user module given that the user has to present a token tied to its id, so in this case "there is a user"

Reply:
LGTM. At this point we are already dealing with the user so makes sense to me keeping the method here.